core/terminal.go

```diff
@@ -1,44 +1,21 @@
 package core
 
 import (
-	"bytes"
 	"context"
 	"errors"
 	"fmt"
 	"io"
-	"net/http"
-	"strconv"
-	"strings"
 
-	"github.com/gorilla/websocket"
 	bkgw "github.com/moby/buildkit/frontend/gateway/client"
 	bkgwpb "github.com/moby/buildkit/frontend/gateway/pb"
-	"github.com/moby/buildkit/util/bklog"
-	"github.com/vektah/gqlparser/v2/ast"
+	"github.com/muesli/termenv"
 	"golang.org/x/sync/errgroup"
 
 	"github.com/dagger/dagger/dagql/call"
-	"github.com/dagger/dagger/engine"
+	"github.com/dagger/dagger/dagql/idtui"
 )
 
-type Terminal struct {
-	Endpoint string `json:"endpoint"`
-}
-
-func (*Terminal) Type() *ast.Type {
-	return &ast.Type{
-		NamedType: "Terminal",
-		NonNull:   true,
-	}
-}
-
-func (*Terminal) TypeDescription() string {
-	return "An interactive terminal that clients can connect to."
-}
-
-func (term *Terminal) WebsocketURL() string {
-	return fmt.Sprintf("ws://dagger/%s", term.Endpoint)
-}
+const terminalPrompt = `PS1=\\033[38;5;11mdagger:\w$ \\033[0m`
```

Comment:
nit: a comment on what this actually looks like would help for non-native tty code speakers 😄

Reply:
@vito Are there TUI utilities for coloring I could reuse instead of manual escapes?

Reply:
`termenv` is pretty good for this - we use a lot of it in `frontend.go` fyi.

Reply:
Thank you @jedevc! Updated